### PR TITLE
Add tests for new System.Reflection APIs

### PR DESCRIPTION
**Pull request by sepidehMS**
Adding tests for new System.Reflection APIs in netstandard1.7. 
[Part of these tests have been ported from ProjectN and ProjectK branches]

CC: @danmosemsft @weshaggard @AtsushiKan 
FYI: @alexperovich

### CODE REVIEW

File: src/System.Runtime/tests/System/Reflection/TypeTests.GetMember.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+using System;
+using System.IO;
+using System.Text;
+using System.Reflection;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Xunit;
+
+#pragma warning disable 0649
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/Reflection/TypeTests.GetMember.cs]: >49 [](start = 26, length = 2)

 // field is never assigned to

File: src/System.Runtime/tests/System/Reflection/AssemblyTests.cs [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+        {
+            Assembly assembly = typeof(AssemblyTests).Assembly;
+            Assert.Throws(exceptionType, () => assembly.CreateInstance(typeName, true, BindingFlags.Public, null, null, null, null));
+            Assert.Throws(exceptionType, () => assembly.CreateInstance(typeName, false, BindingFlags.Public, null, null, null, null));
+        }
+
+        [Fact]
+        public static void GetManifestResourceStream()
+        {
+            Assert.NotNull(typeof(AssemblyTests).Assembly.GetManifestResourceStream(typeof(AssemblyTests), "EmbeddedImage.png"));
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/Reflection/AssemblyTests.cs]: >EmbeddedImage [](start = 108, length = 13)

please replace this big image with a 1x1 gif or png

File: src/System.Runtime/tests/System.Runtime.Tests.csproj [hunk 3]
```diff
@@ -119,6 +119,29 @@
...
+    <Compile Include="System\Reflection\ParameterInfoTests.cs" />
+    <Compile Include="System\Reflection\PropertyInfoTests.cs" />
+    <Compile Include="System\Reflection\TypeTests.Get.CornerCases.cs" />
+    <Compile Include="System\Reflection\TypeTests.GetInterface.cs" />
+    <Compile Include="System\Reflection\TypeTests.GetMember.cs" />
+    <Compile Include="System\Reflection\CustomAttributesTestData.cs" />
+    <EmbeddedResource Include="$(MSBuildThisFileDirectory)\System\Reflection\EmbeddedImage.png" >
+      <LogicalName>System.Reflection.Tests.EmbeddedImage.png</LogicalName>
+    </EmbeddedResource>
+    <EmbeddedResource Include="$(MSBuildThisFileDirectory)\System\Reflection\EmbeddedTextFile.txt" >
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: >$(MSBuildThisFileDirectory)\ [](start = 31, length = 28)

this part should be implicit (same as for the source files above)

File: src/System.Runtime/tests/System/Reflection/EventInfoTests.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                {
+                    eventFound = allevents.Current;
+                    break;
+                }
+            }
+            return eventFound;
+        }
+    }
+
+#pragma warning disable 0067
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/Reflection/EventInfoTests.cs]: >0067 [](start = 24, length = 4)

 // the event was declared but never used

(I think it's helpful to note what the disable is for)

File: src/System.Runtime/tests/System/Reflection/MethodBodyTests.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            MethodInfo mi = typeof(MethodBodyTests).GetMethod("MethodBodyExample");
+            MethodBody mb = mi.GetMethodBody();
+
+            Assert.True(mb.InitLocals);  // local variables are initialized
+            Assert.Equal(2, mb.MaxStackSize);
+            Assert.Equal(5, mb.LocalVariables.Count);
+
+            foreach(LocalVariableInfo lvi in mb.LocalVariables)
+            {
+                if(lvi.LocalIndex == 0) { Assert.Equal(typeof(int), lvi.LocalType); }
```
Comment (danmosemsft) [inline src/System.Runtime/tests/System/Reflection/MethodBodyTests.cs]: >f( [](start = 17, length = 2)

nit style, replace "if(" with "if (" ?

**Comment by danmosemsft** [comment]
:shipit:

File: src/System.Runtime/tests/System/Reflection/MethodBodyTests.cs (hunk 5 above)
Comment (sepidehMS) [inline src/System.Runtime/tests/System/Reflection/MethodBodyTests.cs]: Will do.

File: src/System.Runtime/tests/System/Reflection/EventInfoTests.cs (hunk 4 above)
Comment (sepidehMS) [inline src/System.Runtime/tests/System/Reflection/EventInfoTests.cs]: Will do.

File: src/System.Runtime/tests/System.Runtime.Tests.csproj (hunk 3 above)
Comment (sepidehMS) [inline src/System.Runtime/tests/System.Runtime.Tests.csproj]: Will do.

File: src/System.Runtime/tests/System/Reflection/AssemblyTests.cs (hunk 2 above)
Comment (sepidehMS) [inline src/System.Runtime/tests/System/Reflection/AssemblyTests.cs]: Will do.

File: src/System.Runtime/tests/System/Reflection/TypeTests.GetMember.cs (hunk 1 above)
Comment (sepidehMS) [inline src/System.Runtime/tests/System/Reflection/TypeTests.GetMember.cs]: Will do.

**Comment by sepidehMS** [comment]
Any additional comments?

**Comment by AtsushiKan** [comment]
LGTM

**Pull request merged**